doc/ref_arch/kubernetes/chapters/chapter02.md

```diff
@@ -20,20 +20,149 @@
+| `req.lcm.adp.03` | Automated deployment | The Architecture **should** support hitless upgrade of all software provided by the cloud provider that are not covered by `req.lcm.adp.02`. Whenever hitless upgrades are not feasible, attempt should be made to minimize the duration and nature of impact. |
+| `req.lcm.adp.04` | Automated deployment | The Architecture **must** support declarative specifications of hardware and software assets for automated deployment, configuration, maintenance and management. |
+| `req.lcm.adp.05` | Automated deployment | The Architecture **should** support automated process for Deployment and life-cycle management of CIM Instances. |
+| `req.lcm.cid.02` | CI/CD | The Architecture **should** support integration with CI/CD Toolchain for NFVI and CIM components Automation. |
+
+<p align="center"><b>Table 2-5:</b> Kubernetes Architecture: Operations and LCM Requirements </p>
+
 
+<a name="2.3.6"></a>
+### 2.3.6 Assurance
+
+| Ref # | sub-category | Description |
+|----|------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `req.asr.mon.01` | Integration | The Architecture **must** include integration with infrastructure components to support collection of telemetry for assurance monitoring and network intelligence. |
+| `req.asr.mon.02` | Monitoring | The Architecture **should** support Network Intelligence capabilities that allow richer diagnostic capabilities which take as input broader set of data across the network and from CNF workloads. |
+| `req.asr.mon.03` | Monitoring | The Architecture **must** allow for the collection and dissemination of performance and fault information. |
+| `req.asr.mon.04` | Network | The NFVI Network Fabric and Network Operating System **must** provide network operational visibility through alarming and streaming telemetry services for operational management, engineering planning, troubleshooting, and network performance optimisation. |
+
+<p align="center"><b>Table 2-6:</b> Kubernetes Architecture: Assurance Requirements</p>
+
+<a name="2.3.7"></a>
+### 2.3.7 Security
+
+| Ref # | sub-category | Description |
+|----|------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `req.sec.gen.01` | General | The Architecture **should** provide workload isolation. |
+| `req.sec.gen.02` | General | The Architecture **must** support policy based RBAC. |
+| `req.sec.gen.03` | General | The Architecture **must** support a centralised authentication and authorisation mechanism. |
+| `req.sec.gen.04` | General | The Architecture **should** provide Operating System kernel isolation. |
```

Comment:
Can we have another requirement here ?
| `req.sec.gen.08` | General | The Architecture **should** provide Operating System hardening.

Reply:
@msidana Are you suggesting that the NFVI or K8s should provide "Operating System hardening"? Or should this be "The Architecture should utilize hardened Operating Systems." -- would we need to define "hardened"? A more general requirement would be "The Architecture should utilize hardened components and services."

Reply:
I am suggesting that the base OS of k8s nodes should be hardened (something like openstack-ansible does)
To define "hardened", can we state it as ```The Architecture should provide Operating System hardening per STIG.``` ?

Reply:
I would suggest for hardening, we call out the specific things we want to see, e.g.
`The Architecture should minimise the attack surface by ensuring only required services and processes are running on the Node OS.`
But I think any details such as, e.g. file/directory ownership/permissions and so on don't belong in the RA?  Perhaps we could also have a requirement that suggests the architecture should be able to adopt the CIS Benchmarks for Kubernetes?
`The Architecture should support the ability to conform to the CIS Benchmark for Kubernetes` or something to that effect?

Reply:
@tomkivlin Thanks. Added the following -- generalized "file and directory"

| `req.sec.gen.08` | General | The Architecture should minimise the attack surface by ensuring only required services and processes are running on the Node Operating System. |
| `req.sec.gen.09` | General | The Architecture should support the ability to restrict resource, service, capabilities and interface permissions based on the principles of least privilege. |
| `req.sec.gen.10` | General | The Architecture should support the ability to conform to the Center for Internet Security (CIS) Benchmark for Kubernetes. |

Reply:
Great suggestions @tomkivlin .
Thanks @pgoyal01, looks good now :)